ambry-messageformat/src/main/java/com/github/ambry/messageformat/MessageFormatRecord.java

```diff
@@ -1756,6 +1754,78 @@ public static BlobData deserializeBlobRecord(CrcInputStream crcStream) throws IO
     }
   }
 
+  /**
+   *  - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   * |         |           |              |            |            |            |
+   * | version | blobType  | isCompressed |    size    |  content   |     Crc    |
+   * |(2 bytes)| (2 bytes) | (1 byte)     |  (8 bytes) |  (n bytes) |  (8 bytes) |
+   * |         |           |              |            |            |            |
+   *  - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   *  version    - The version of the blob record
+   *
+   *  blobType   - The type of the blob
+   *
+   *  isCompressed - Whether the content is compressed.
+   *
+   *  size       - The size of the blob content
+   *
+   *  content    - The actual content that represents the blob
+   *
+   *  crc        - The crc of the blob record
+   *
+   */
+  public static class Blob_Format_V3 {
+    public static final int Blob_Size_Field_In_Bytes = 8;
+    public static final int Blob_Type_Field_In_Bytes = 2;
+    public static final int Is_Compressed_In_Bytes = 1;
+
+    private static Logger logger = LoggerFactory.getLogger(Blob_Format_V3.class);
+
+    public static long getBlobRecordSize(long blobSize) {
+      return Version_Field_Size_In_Bytes + Blob_Type_Field_In_Bytes + Blob_Size_Field_In_Bytes + blobSize + Crc_Size
+          + Is_Compressed_In_Bytes;
+    }
+
+    public static void serializePartialBlobRecord(ByteBuffer outputBuffer, long blobContentSize, BlobType blobType,
+        boolean isCompressed) {
+      outputBuffer.putShort(Blob_Version_V3);
+      outputBuffer.putShort((short) blobType.ordinal());
+      outputBuffer.put(isCompressed ? (byte) 1 : 0);
+      outputBuffer.putLong(blobContentSize);
+    }
+
+    public static BlobData deserializeBlobRecord(CrcInputStream crcStream) throws IOException, MessageFormatException {
+      DataInputStream dataStream = new DataInputStream(crcStream);
+      // Get the BlobType.
+      short blobTypeOrdinal = dataStream.readShort();
+      if (blobTypeOrdinal > BlobType.values().length) {
+        logger.error("corrupt data while parsing blob content BlobContentType {}", blobTypeOrdinal);
```

Comment:
Should we have ">=" instead of ">" here?

Reply:
You are right.  It should be >=, let me correct that.